tests/integration/sync_test.go

```diff
@@ -3294,258 +3956,667 @@ func Test_Sync_ChangingIDsWhileKeepingNames(t *testing.T) {
+				assert.NoError(t, err)
+				defer resp.Body.Close()
+				if resp.StatusCode == http.StatusTooManyRequests {
+					break
+				}
+				n++
+			}
+			assert.Equal(t, maxRegularRequestsNumber, n)
+		})
+	}
 }
 
 // test scope:
-//   - 3.0.0+
-//   - konnect
-func Test_Sync_CreateCertificateWithSNIs(t *testing.T) {
-	runWhenKongOrKonnect(t, ">=3.0.0")
-
-	client, err := getTestClient()
-	if err != nil {
-		t.Errorf(err.Error())
+//   - > 3.4.0
+func Test_Sync_ConsumerGroupsScopedPlugins_Post340(t *testing.T) {
+	tests := []struct {
+		name          string
+		kongFile      string
+		expectedError error
+	}{
+		{
+			name:     "attempt to createe consumer groups scoped plugins with older Kong version",
+			kongFile: "testdata/sync/017-consumer-groups-rla-application/kong3x.yaml",
+			expectedError: errors.New(
+				"building state: a rate-limiting-advanced plugin with config.consumer_groups\n" +
+					"and/or config.enforce_consumer_groups was found. Please use Consumer Groups scoped\n" +
+					"Plugins when running against Kong Enterprise 3.4.0 and above.\n\n" +
+					"Check DOC_LINK for more information"),
+		},
 	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			runWhen(t, "enterprise", ">=3.4.0")
+			teardown := setup(t)
+			defer teardown(t)
 
-	err = sync("testdata/sync/023-create-and-update-certificate-with-snis/initial.yaml")
-	require.NoError(t, err)
-
-	// To ignore noise, we ignore the Key and Cert fields because they are not relevant for this test.
-	ignoredFields := []cmp.Option{
-		cmpopts.IgnoreFields(
-			kong.Certificate{},
-			"Key",
-			"Cert",
-		),
+			err := sync(tc.kongFile)
+			assert.EqualError(t, err, tc.expectedError.Error())
+		})
 	}
-
-	testKongState(t, client, false, utils.KongRawState{
-		Certificates: []*kong.Certificate{
-			{
-				ID:   kong.String("c75a775b-3a32-4b73-8e05-f68169c23941"),
-				Tags: kong.StringSlice("before"),
-			},
-		},
-		SNIs: []*kong.SNI{
-			{
-				Name: kong.String("example.com"),
-				Certificate: &kong.Certificate{
-					ID: kong.String("c75a775b-3a32-4b73-8e05-f68169c23941"),
-				},
-			},
-		},
-	}, ignoredFields)
-
-	err = sync("testdata/sync/023-create-and-update-certificate-with-snis/update.yaml")
-	require.NoError(t, err)
-
-	testKongState(t, client, false, utils.KongRawState{
-		Certificates: []*kong.Certificate{
-			{
-				ID:   kong.String("c75a775b-3a32-4b73-8e05-f68169c23941"),
-				Tags: kong.StringSlice("after"), // Tag should be updated.
-			},
-		},
-		SNIs: []*kong.SNI{
-			{
-				Name: kong.String("example.com"),
-				Certificate: &kong.Certificate{
-					ID: kong.String("c75a775b-3a32-4b73-8e05-f68169c23941"),
-				},
-			},
-		},
-	}, ignoredFields)
 }
 
-// test scope:
-//   - 3.0.0+
-//   - konnect
-func Test_Sync_ConsumersWithCustomIDAndUsername(t *testing.T) {
-	runWhenKongOrKonnect(t, ">=3.0.0")
+func Test_Sync_ConsumerGroupsScopedPluginsKonnect(t *testing.T) {
+	// TODO: remove skip once Konnect support is enabled.
```

Comment:
Does it mean Konnect doesn't support plugin-scoped ConsumerGroups yet? If so we'll have to account for this in KIC somehow. 🤔  @programmer04

Reply:
plugin-scoped ConsumerGroups support in 3.4 is not released yet, so no, this is not supported in Konnect.